ddtrace/_trace/tracer.py

```diff
@@ -232,7 +233,18 @@ def __init__(
         self.context_provider = context_provider or DefaultContextProvider()
         # _user_sampler is the backup in case we need to revert from remote config to local
         self._user_sampler: Optional[BaseSampler] = DatadogSampler()
+        self._asm_enabled = asm_config._asm_enabled
         self._sampler: BaseSampler = DatadogSampler()
+        self._apm_opt_out = self._asm_enabled and not self.enabled
+        if self._apm_opt_out:
+            # If ASM is enabled but tracing is disabled,
+            # we need to set the rate limiting to 1 trace per minute
+            # for the backend to consider the service as alive.
+            from ddtrace.internal.rate_limiter import RateLimiter
+
+            self._sampler.limiter = RateLimiter(rate_limit=1, time_window=60e9)  # 1 trace per minute
+            # Disable compute stats
+            config._trace_compute_stats = False
```

Comment:
Why do we need to disable compute stats?

Reply:
Well, it's part of the RFC. I think because we're disabling "APM" features basically